lib/api/apiUtils/object/replication.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Get the object metadata replication information
+ * @param {string} config - The bucket replication configuration
+ * @param {object} rule - The replication rule
+ * @param {array} content - The array of content types to be replicated
+ * @return {object} The object metadata replication information
+ */
+function _getReplicationInfo(config, rule, content) {
+    const replicationInfo = {
+        status: 'PENDING',
+        content,
+        destination: config.destination,
+    };
+    // StorageClass is optional for rules.
+    if (rule.storageClass) {
+        Object.assign(replicationInfo, { storageClass: rule.storageClass });
+    }
+    return replicationInfo;
+}
+
+/**
+ * Check that a bucket replication rule applies for the given object key. If it
+ * does, assign replication information to the given object.
+ * @param {object} obj - The object to set replicationInfo for
+ * @param {string} objKey - The key of the object
+ * @param {object} bucketMD - The bucket metadata
+ * @param {boolean} isDeleteMarker - Whether the object is a deleteMarker
+ * @return {undefined}
+ */
+function assignObjectReplicationInfo(obj, objKey, bucketMD, isDeleteMarker) {
```

Comment:
I've renamed the main function `buildReplicationInfo`, but have kept the file name since I've added two wrapper functions (see next comment).